python/cudf/cudf/core/groupby/groupby.py

```
@@ -780,6 +784,126 @@ def median(self):
+        6  c     4     8     8
+        7  c     5     5     5
+        8  c     2     1     1
+        >>> gdf.groupby("id").corr(method="pearson")
+                    val1      val2      val3
+        id
+        a   val1  1.000000  0.500000  0.500000
+            val2  0.500000  1.000000  1.000000
+            val3  0.500000  1.000000  1.000000
+        b   val1  1.000000  0.385727  0.385727
+            val2  0.385727  1.000000  1.000000
+            val3  0.385727  1.000000  1.000000
+        c   val1  1.000000  0.714575  0.714575
+            val2  0.714575  1.000000  1.000000
+            val3  0.714575  1.000000  1.000000
+
+        """
+
+        if not method.lower() in ["pearson"]:
+            raise NotImplementedError(
+                "Only pearson correlation is currently supported"
+            )
+
+        # create expanded dataframe consisting all combinations of the
+        # struct columns-pairs to be correlated
+        # i.e (('col1', 'col1'), ('col1', 'col2'), ('col2', 'col2'))
+        _cols = self.grouping.values.columns.tolist()
+
+        if self._by:
+            new_df = cudf.DataFrame._from_data(self.grouping.keys._data)
+            new_df._data.multiindex = False
+        else:
+            new_df = cudf.DataFrame._from_data(
+                {}, index=_index_from_data(self.grouping.keys._data)
+            )
+
+        for i in tuple(itertools.combinations_with_replacement(_cols, 2)):
+            new_df._data[i] = cudf.DataFrame._from_data(
+                {"x": self.obj._data[i[0]], "y": self.obj._data[i[1]]}
+            ).to_struct()
+        new_gb = new_df.groupby(by=self._by, level=self._level)
+
+        try:
+            gb_corr = new_gb.agg(lambda x: x.corr(method, min_periods))
+        except RuntimeError as e:
+            if "Unsupported groupby reduction type-agg combination" in str(e):
+                raise TypeError(
+                    "Correlation accepts only numerical column-pairs"
+                ) from e
```

I don't think we want the `from e` here. We don't want to surface the exception from the Cython API.
```suggestion
                )
```